src/GitVersionExe/IGitVersionExecutor.cs

```diff
@@ -0,0 +1,7 @@
+namespace GitVersion
+{
+    public interface IGitVersionExecutor
+    {
+        int Execute(Arguments arguments);
```

Comment:
I think it's hard to understand the relationship between `IGitVersionExecutor`, `IExecCommand` and `IGitVersionExecutor`. Are all three interfaces and corresponding implementations necessary?

Reply:
I agree we should refactor this. I think it's out of scope of this PR, The main goal for this PR was to introduce DI, and the abstractions we can work with later, do you mind leaving it for another PR?